astropy/cosmology/tests/test_connect.py

```diff
@@ -63,31 +77,35 @@ def test_readwrite_complete_info(self, cosmo, tmp_path, format):
         got = Cosmology.read(fname, format=format)
 
         assert got == cosmo
-        assert dict(got.meta) == dict(cosmo.meta)
+        assert (not metaio) ^ (dict(got.meta) == dict(cosmo.meta))
```

Comment:
Is it guaranteed that the metadata do not match if `not metaio`? My tendency would be `if metaio: assert ...`

Reply:
It should be guaranteed. And if it ever starts failing that means the metadata is supported, which would be excellent. This test can't become out of date.